### ISSUE TITLE
DefaultConverterProvider should be Serializable

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10839]
An oversight, `DefaultConverterProvider` isn't `Serializable`

This unit test currently fails:

```java
@Test
public void testSerializableConverterProvider() throws Exception {
    Table1Record r1 = create.newRecord(TABLE1);
    r1.set(FIELD_ID1, 1);
    r1 = runSerialisation(r1);
    assertEquals("1", r1.get(FIELD_ID1, String.class));
}
```

The failure is:

```
java.lang.NullPointerException
	at org.jooq.impl.Tools.converter(Tools.java:1133)
	at org.jooq.impl.Tools.converterOrFail(Tools.java:1162)
	at org.jooq.impl.Tools.converterOrFail(Tools.java:1181)
	at org.jooq.impl.AbstractRecord.get(AbstractRecord.java:262)
	at org.jooq.test.SerializationTest.testSerializableConverterProvider(SerializationTest.java:99)
```

----
See also:
- https://github.com/jOOQ/jOOQ/issues/10838

**Comment by lukaseder** [comment]
This fixes the NPE for the `DefaultConverterProvider` implementation, but the NPE should also be fixed for custom implementations.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0. Backport scheduled for 3.14.2 (#10840)
